compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java

```diff
@@ -605,13 +610,67 @@ private void genHandlingBlockedOnExternal(int localVarOffset, Label blockedOnExt
     private void storeReturnFromCallIns(BIRNode.BIRVariableDcl lhsOpVarDcl) {
 
         if (lhsOpVarDcl != null) {
+            this.genCastToLhsType(lhsOpVarDcl);
             this.storeToVar(lhsOpVarDcl);
         } else {
             this.mv.visitInsn(POP);
         }
     }
 
-        private void genCall(BIRTerminator.Call callIns, String orgName, String moduleName,
```

Comment:
can we use generateCast method in JVMCastGen

Reply:
I couldnt find any method in JVMCastGen which will do this. Any pointers would help.

Reply:
this will add cast to all the method calls. I think issue was with only old externs functions.